test/gtest/glu.hpp

```diff
@@ -0,0 +1,289 @@
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ *
+ *******************************************************************************/
+
+#include "../driver/tensor_driver.hpp"
+#include "cpu_glu.hpp"
+#include "get_handle.hpp"
+#include "miopen/allocator.hpp"
+#include "random.hpp"
+#include "tensor_holder.hpp"
+#include "verify.hpp"
+#include <algorithm>
+#include <cmath>
+#include <gtest/gtest.h>
+#include <miopen/miopen.h>
+#include <miopen/glu.hpp>
+#include <numeric>
+
+struct GLUTestCase
+{
+    size_t N;
+    size_t C;
+    size_t D;
+    size_t H;
+    size_t W;
+    int32_t dim;
+    friend std::ostream& operator<<(std::ostream& os, const GLUTestCase& tc)
+    {
+        return os << " N:" << tc.N << " C:" << tc.C << " D:" << tc.D << " H:" << tc.H
+                  << " W:" << tc.W << " dim:" << tc.dim;
+    }
+
+    std::vector<size_t> GetInput()
+    {
+        if((N != 0) && (C != 0) && (D != 0) && (H != 0) && (W != 0))
+        {
+            return std::vector<size_t>({N, C, D, H, W});
+        }
+        else if((N != 0) && (C != 0) && (H != 0) && (W != 0))
+        {
+            return std::vector<size_t>({N, C, H, W});
+        }
+        else if((N != 0) && (C != 0) && (W != 0))
+        {
+            return std::vector<size_t>({N, C, W});
+        }
+        else if((N != 0) && (W != 0))
+        {
+            return std::vector<size_t>({N, W});
+        }
+        else if((N != 0))
+        {
+            return std::vector<size_t>({N});
+        }
+        else
+        {
+            std::cout << "Error Input Tensor Lengths\n" << std::endl;
+            return std::vector<size_t>({0});
+        }
+    }
+};
+
+std::vector<GLUTestCase> GLUTestConfigs()
+{ // n c d h w dim
+    // clang-format off
+    return {
+        { 2,    320,   4,  4,   4,    0},
+        { 32,    64,   3,  3,    3,     0},
+        { 64,    3,  0,  11,    11,     0},
+        { 256,    256,  0,  1,    1,     0},
+        { 128,    64,  0,  7,    7,     0},
```

Comment:
Do you mean excluding the test case where total elements exceed 400,000 or excluding the condition `if(!(input_numel < 400000)) return false;` ?

Reply:
```
[----------] Global test environment tear-down
[==========] 24 tests from 3 test suites ran. (572 ms total)
[  PASSED  ] 21 tests.
[  FAILED  ] 3 tests, listed below:
[  FAILED  ] GLUTestSet/GLUFwdTestFloat.GLUTestFw/4, where GetParam() =  N:128 C:64 D:0 H:7 W:7 dim:0
[  FAILED  ] GLUTestSet/GLUFwdTestFP16.GLUTestFw/4, where GetParam() =  N:128 C:64 D:0 H:7 W:7 dim:0
[  FAILED  ] GLUTestSet/GLUFwdTestBFP16.GLUTestFw/4, where GetParam() =  N:128 C:64 D:0 H:7 W:7 dim:0
```
The current test is failing and needs to be turned into a success.

Is the distinction of 400,000 in IsApplicable() due to performance reasons? 
If so, then IsApplicable() shouldn't be modified. 
Please modify the test code so that the test ends successfully.

You have the following options to choose from:

1. Excluding ```{ 128, 64, 0, 7, 7, 0}.```
2. Expecting return false when the size is greater than 400,000.
```
status = miopen::GLUForward(handle, input.desc, input_dev.get(), dim, output.desc, output_dev.get());
if (input.data.size() >= 400000) 
    EXPECT_NE(status, miopenStatusSuccess);
else 
    EXPECT_EQ(status, miopenStatusSuccess);
```
3. Or any better alternative method you might have in mind.

Reply:
I have excluded the test cases that don't pass isApplicable()